File: MB-Whatsapp/production-safety/startup-validator.ts
import { validateProductionEnv } from "./env-validator";
import { buildProductionSafetyHealth } from "./health-check";
import { evaluateRuntimeGuard, loadProductionSafetyConfig } from "./runtime-guard";
import type { StartupValidationResult } from "./types";

export function validateProductionSafetyStartup(env: NodeJS.ProcessEnv = process.env): StartupValidationResult {
  const config = loadProductionSafetyConfig(env);
  const guard = evaluateRuntimeGuard(config);
  const envValidation = validateProductionEnv(env);
  const errors = [...envValidation.dangerous_flags];
  const warnings = [...envValidation.warnings];

  return {
    ok: errors.length === 0,
    errors,
    warnings,
    guard,
    health: buildProductionSafetyHealth(config),
    created_at: new Date().toISOString(),
  };
}

File: MetaBrain/production_safety_py/startup_validator.py
from datetime import UTC, datetime
from collections.abc import Mapping

from .env_validator import validate_production_env
from .health_check import build_production_safety_health
from .runtime_guard import evaluate_runtime_guard, load_production_safety_config
from .types import StartupValidationResult


def validate_production_safety_startup(env: Mapping[str, str]) -> StartupValidationResult:
    config = load_production_safety_config(env)
    guard = evaluate_runtime_guard(config)
    env_validation = validate_production_env(env)
    return StartupValidationResult(
        ok=not env_validation.dangerous_flags,
        errors=list(env_validation.dangerous_flags),
        warnings=list(env_validation.warnings),
        guard=guard,
        health=build_production_safety_health(config),
        created_at=datetime.now(UTC).isoformat(),
    )
